keylime-agent/src/main.rs

```diff
@@ -497,18 +567,49 @@ async fn main() -> Result<()> {
 
     {
         // Request keyblob material
-        let keyblob = registrar_agent::do_register_agent(
-            config.agent.registrar_ip.as_ref(),
-            config.agent.registrar_port,
-            &agent_uuid,
-            &PublicBuffer::try_from(ek_result.public.clone())?.marshall()?,
-            ek_result.ek_cert,
-            &PublicBuffer::try_from(ak.public)?.marshall()?,
-            mtls_cert,
-            config.agent.contact_ip.as_ref(),
-            config.agent.contact_port,
-        )
-        .await?;
+        let keyblob = if config.agent.enable_iak_idevid {
+            registrar_agent::do_register_agent(
+                config.agent.registrar_ip.as_ref(),
+                config.agent.registrar_port,
+                &agent_uuid,
+                &PublicBuffer::try_from(ek_result.public.clone())?
+                    .marshall()?,
+                ek_result.ek_cert,
+                &PublicBuffer::try_from(ak.public)?.marshall()?,
+                Some(
+                    &PublicBuffer::try_from(iak.unwrap().public.clone())? //#[allow_ci]
+                        .marshall()?,
+                ),
+                Some(
+                    &PublicBuffer::try_from(idevid.unwrap().public.clone())? //#[allow_ci]
+                        .marshall()?,
+                ),
+                Some(attest.unwrap().marshall()?), //#[allow_ci]
+                Some(signature.unwrap().marshall()?), //#[allow_ci]
```

Comment:
Please, avoid using `unwrap()` here.  You could use matching to "extract" the contents of the Options before calling `do_register_agent`.

A suggestion would be to use something like:

```
if let (Some(iak), Some(idevid), Some(attest), Some(signature)) = (iak, idevid, attest, signature)  {
    registrar_agent::do_register_agent(...)
} else {
    error!(...)
}
```

I did not test this, but I think it would avoid using unwrap here

Reply:
I have removed the unwraps as suggested